libsrc/cbm/cbm_read.s

```diff
@@ -92,9 +92,9 @@ _cbm_read:
         bne     @L3
         inc     ptr3+1          ; ++bytesread;
 
-@L3:    inc     ptr1
+@L3:    dec     ptr1
```

Comment:
What does your code do for `cbm_read(_, _, 0);`? It looks like the first test decrements from 0 to 2¹⁶ − 1, so it will loop 2¹⁶ times.

Reply:
> What does your code do for `cbm_read(_, _, 0);`? It looks like the first test decrements from 0 to 2¹⁶ − 1, so it will loop 2¹⁶ times.
Nope. $0000 becomes $0101 at init and will exit immediately. I'll have to print out this thread and frame it because now I'm sure that I'm not alone with the way I was feeling about this hack.

Reply:
At the beginning of the function, the two bytes are incremented, separately, to `$0101`.  The loop starts by decrementing the low byte to `$00`.  Then, the high byte is decremented to `$00`.  The loop immediately stops -- it reads nothing.

Reply:
`ptr1+0` and `ptr1+1` aren't used as a 16-bit counter; they're two 8-bit counters.  `ptr1+0` counts a loop that's nested inside of a second loop that's counted by `ptr1+1`.

Reply:
I mis-parsed the comment and forgot about the increment when looking at the loop guard. The new comment is much better. Nice hack!